src/common/commonutils/DeviceInfoUtils.c

```diff
@@ -840,6 +840,18 @@ bool IsRedHatBased(OsConfigLogHandle log)
     return result;
 }
 
+bool IsRedHatBased(OsConfigLogHandle log)
+{
+    static bool firstTime = true;
+    static bool redHatBased = false;
```

Comment:
Can we instead just call this?
```suggestion
    static bool redHatBased = IsRedHatBasedInternal(log);
```

Reply:
Tried that but it's not allowed in C:
```
.../src/common/commonutils/DeviceInfoUtils.c: In function ‘IsRedHatBased’:
.../src/common/commonutils/DeviceInfoUtils.c:845:31: error: initializer element is not constant
  845 |     static bool redHatBased = IsRedHatBasedInternal(log);
      |                               ^~~~~~~~~~~~~~~~~~~~~
```